File: isAnagram.swift
class Solution {
    func isAnagram(_ s: String, _ t: String) -> Bool {
        var seen: [Character: Int] = [:]
        for char in s {
            seen[char] = seen[char, default: 0] + 1
        }
        for char in t {
            if seen[char] == nil {return false}
            seen[char] = seen[char]! - 1
        }
        for key in seen.keys {
            if seen[key] != 0 {return false}
        }
        return true
    }
}

File: isAnagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        counter = {}
        for char in s:
            counter[char] = counter.get(char, 0) + 1
        for char in t:
            if char not in counter: return False
            counter[char] -= 1
        for count in counter.values():
            if count != 0:
                return False
        return True
